deploy/example/rook-ceph/cluster-up.sh

```diff
@@ -0,0 +1,256 @@
+  echoerr "ERROR: The --subscription option is required."
+  printhelp
+  exit 1
+fi
+
+if [[ -z ${AZURE_LOCATION:-} ]]; then
+  echoerr "ERROR: The --location option is required."
+  printhelp
+  exit 1
+fi
+
+if [[ -z ${AZURE_CLUSTER_TEMPLATE:-} ]]; then
+  AZURE_CLUSTER_TEMPLATE="multi-az"
+fi
+
+if [[ -z ${AZURE_K8S_VERSION:-} ]]; then
+  AZURE_K8S_VERSION="1.18"
+fi
+
+IS_AZURE_CLUSTER_TEMPLATE_URI=$(expr "$(expr "${AZURE_CLUSTER_TEMPLATE}" : "file://\|https://\|http://")" != 0 || true)
+
+if [[ ${IS_AZURE_CLUSTER_TEMPLATE_URI} -eq 0 ]]; then
+  AZURE_CLUSTER_TEMPLATE_ROOT=${GIT_ROOT}/deploy/example/rook-ceph/cluster
+  AZURE_CLUSTER_TEMPLATE_FILE=${AZURE_CLUSTER_TEMPLATE_ROOT}/${AZURE_CLUSTER_TEMPLATE}/aks-engine-config.json
+
+  if [[ ! -f "$AZURE_CLUSTER_TEMPLATE_FILE" ]]; then
+    AZURE_CLUSTER_VALID_TEMPLATES=$(find "${AZURE_CLUSTER_TEMPLATE_ROOT}" -maxdepth 1 -printf "%P\n" | awk '{split($1,f,"."); printf (NR>1?", ":"") f[1]}')
+    echoerr "ERROR: The template '$AZURE_CLUSTER_TEMPLATE' is not known. Select one of the following values: $AZURE_CLUSTER_VALID_TEMPLATES."
+    printhelp
+    exit 1
+  fi
+
+  AZURE_CLUSTER_TEMPLATE_FILE=file://${AZURE_CLUSTER_TEMPLATE_FILE}
+else
+  AZURE_CLUSTER_TEMPLATE_FILE=${AZURE_CLUSTER_TEMPLATE}
+fi
+
+if [[ -z ${AZURE_CLUSTER_DNS_NAME:-} ]]; then
+  CLUSTER_PREFIX=$(whoami)
+  if [[ ${CLUSTER_PREFIX:-root} == "root" ]]; then
+    CLUSTER_PREFIX=k8s
+  fi
+  AZURE_CLUSTER_DNS_NAME=$(basename "$(mktemp -t "${CLUSTER_PREFIX}-${AZURE_CLUSTER_TEMPLATE}-${GIT_COMMIT}-XXX")")
+fi
+
+if [[ -z ${AZURE_RESOURCE_GROUP:-} ]]; then
+  AZURE_RESOURCE_GROUP=${AZURE_CLUSTER_DNS_NAME}-rg
+fi
+
+if [[ -z ${OUTPUT_DIR:-} ]]; then
+  OUTPUT_DIR="$GIT_ROOT/_output/$AZURE_CLUSTER_DNS_NAME"
+fi
+
+#
+# Install required tools
+#
+install_helm
+
+#
+# Create the Kubernetes cluster
+#
+echo "Creating cluster ${AZURE_CLUSTER_DNS_NAME}"
+"${GIT_ROOT}/hack/azure-cluster-up.sh" \
+  --subscription "${AZURE_SUBSCRIPTION_ID}" \
+  --location "${AZURE_LOCATION}" \
+  --name "${AZURE_CLUSTER_DNS_NAME}" \
+  --output "${OUTPUT_DIR}" \
+  --k8s-version "${AZURE_K8S_VERSION}" \
+  --template "${AZURE_CLUSTER_TEMPLATE_FILE}"
+
+# Delete the cluster on subsequent errors.
+trap_push "\"${OUTPUT_DIR}/cluster-down.sh\"" err
+
+echo "Wait for cluster to become available..."
+# TODO Figure out a better way to determine cluster availability
+sleep 5m
+
+export KUBECONFIG="${OUTPUT_DIR}/kubeconfig/kubeconfig.${AZURE_LOCATION}.json"
+
+#
+# Apply taints and labels to the storage pool nodes.
+#
+echo "Labelling and tainting storage pool nodes..."
+kubectl label nodes --selector agentpool=storagepool storage-node=true
+kubectl taint nodes --selector agentpool=storagepool storage-node:NoSchedule
+
+#
+# Install the Azure Disk CSI Driver
+#
+echo "Installing Azure Disk CSI Driver..."
+helm repo add azuredisk-csi-driver https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/master/charts
```

Comment:
TODO: Install the driver from the current repo and allow for a specific version to be specified on the command line.